joeynmt/training.py

```diff
@@ -291,14 +304,29 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
                 # see https://medium.com/@davidlmorton/
                 # increasing-mini-batch-size-without-increasing-
                 # memory-6794e10db672
+
+                # Set current_batch_mutliplier to fit number of leftover examples for last batch in epoch
+                if self.batch_multiplier > 1 and batch_mutliplier_problem and i == len(train_iter) - math.ceil(leftover_batch_size / self.batch_size):
```

Comment:
I think we can simplify this. We only need to check whether the batch_multiplier is used and whether it's the last batch. Then we retrieve its size and scale the batch_multiplier accordingly. If there is no problem, the scaling should return 1.

Reply:
I deleted the "batch_multiplier_problem' computation, since it was redundant. 368b6b0
I don't think we can simplify it further since we need the number of leftover examples (leftover_batch_size) anyway to get the last actual batch (batch_size * batch_mutliplier).
Scaling self.batch_multiplier itself, instead of computing self.current_batch_mutliplier, isn't sensible since it would than be incorrect for the next epoch.

Reply:
Yes, you're right. Thanks!